src/search_bar.rs

```diff
@@ -63,6 +59,16 @@ impl SearchBar {
             return;
         }
 
+        if edit_response.changed() || self.is_first_frame {
```

Comment:
Is the `is_first_frame` flag what ensures that without any input text, we see ALL gadgets in the gadget bar? If so I'd like a comment explaining that, wasn't obvious to me

Reply:
I could do either that, or move the matching logic to another function, and update it when we call `toggle()`

Reply:
Update: that is exactly what I did :P